edgexfoundry/app-functions-sdk-go
# feat: Location of client service obtained from the registry

Replaces local clients bootstrap handler with new common clients bootstrap handler from go-mod-bootstrap

closes #1032

This PR is dependent on edgexfoundry/go-mod-bootstrap#305 and is in draft mode until it is merged.

Signed-off-by: Leonard Goodell <leonard.goodell@intel.com>







**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/app-functions-sdk-go/blob/main/.github/CONTRIBUTING.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] I have added unit tests for the new feature or bug fix (if not, why?) **N/A - unit tests in module**
      <link to docs PR>

## Testing Instructions
Build app template from this branch
Run app template in with `-r` and `EDGEX_SECURITY_SECRET_STORE=false` and `WRITABLE_LOGLEVEL=DEBUG`
Verify logs contain following messages 
```
level=DEBUG ts=2022-02-02T15:55:23.4001212Z app=new-app-service source=clients.go:148 msg="Using registry for URL for 'core-metadata': http://edgex-core-metadata:59881"
level=DEBUG ts=2022-02-02T15:55:23.4040108Z app=new-app-service source=clients.go:148 msg="Using registry for URL for 'support-notifications': http://edgex-support-notifications:59860"
level=DEBUG ts=2022-02-02T15:55:23.4048862Z app=new-app-service source=clients.go:148 msg="Using registry for URL for 'core-command': http://edgex-core-command:59882"
level=DEBUG ts=2022-02-02T15:55:23.4056781Z app=new-app-service source=clients.go:148 msg="Using registry for URL for 'core-data': http://edgex-core-data:59880"
```
ReRun app template without `-r`
Verify logs contain following messages 
```
level=DEBUG ts=2022-02-02T15:56:23.6967468Z app=new-app-service source=clients.go:126 msg="Using configuration for URL for 'core-metadata': http://localhost:59881"
level=DEBUG ts=2022-02-02T15:56:23.7053054Z app=new-app-service source=clients.go:126 msg="Using configuration for URL for 'support-notifications': http://localhost:59860"
level=DEBUG ts=2022-02-02T15:56:23.705394Z app=new-app-service source=clients.go:126 msg="Using configuration for URL for 'core-command': http://localhost:59882"
level=DEBUG ts=2022-02-02T15:56:23.7054449Z app=new-app-service source=clients.go:126 msg="Using configuration for URL for 'core-data': http://localhost:59880"
```
## New Dependency Instructions (If applicable)
**N/A**
